astropy/utils/data_info.py

```diff
@@ -44,6 +44,10 @@
                      (True, 'S'): 'bytes',  # PY3
                      (True, 'U'): 'str'}
 
+# Context types for which multidimensional data will be flattened to
+# a list of 1-d arrays.
+FLATTEN_MULTIDIM = ['ecsv']
```

Comment:
Not clear if there will ever be another context that supports N-d serialization by flattening, but this seemed better than hardwiring `'ecsv'` in the code.